consumer/heartbeats_consumer.go

```diff
@@ -0,0 +1,315 @@
+package consumer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	// "github.com/RedHatInsights/insights-results-aggregator/producer"
+	"github.com/RedHatInsights/insights-results-aggregator/storage"
+	"github.com/rs/zerolog/log"
+
+	// "github.com/RedHatInsights/insights-results-aggregator/types"
+	"github.com/Shopify/sarama"
+
+	"github.com/RedHatInsights/insights-results-aggregator/broker"
+)
+
+// Metadata taken from https://github.com/redhatinsights/insights-ingress-go/internal/validators/types.go
+type Metadata struct {
```

Comment:
FYI: you don't need to define all the fields of the JSON, just the ones you are interested in. This way if it ever removes a field you don't care about, your implementation won't be broken :)

Reply:
no worries, it was just a note for your Go knowledge :D just saying you don't need to define all the fields